### PR TITLE
dep: update deps that were reported in OSV issues

### PR DESCRIPTION
**Pull request by j75689**
### Description
This pr updates some dependencies to newer versions.

### Rationale
Some dependency was found in OSV-reported issues. update them for security.
refer: https://github.com/google/osv-scanner
```
╭───────────────────────────────────────────────────┬───────────┬─────────────────────────────────────┬───────────────────────────────────┬──────────────────────────╮ ≈
│ OSV URL (ID IN BOLD)                              │ ECOSYSTEM │ PACKAGE                             │ VERSION                           │ SOURCE                   │
├───────────────────────────────────────────────────┼───────────┼─────────────────────────────────────┼───────────────────────────────────┼──────────────────────────┤ ≈
│ https://osv.dev/vulnerability/GHSA-2chg-86hq-7w38 │ Go        │ github.com/btcsuite/btcd            │ 0.0.0-20190115013929-ed77733ec07d │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2022-1098        │           │                                     │                                   │                          │
│ https://osv.dev/vulnerability/GO-2022-0322        │ Go        │ github.com/prometheus/client_golang │ 0.9.1                             │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2021-0227        │ Go        │ golang.org/x/crypto                 │ 0.0.0-20200622213623-75b288015ac9 │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2022-0968        │ Go        │ golang.org/x/crypto                 │ 0.0.0-20200622213623-75b288015ac9 │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2021-0356        │ Go        │ golang.org/x/crypto                 │ 0.0.0-20200622213623-75b288015ac9 │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2022-0236        │ Go        │ golang.org/x/net                    │ 0.0.0-20201021035429-f5854403a974 │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2021-0238        │ Go        │ golang.org/x/net                    │ 0.0.0-20201021035429-f5854403a974 │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2022-0288        │ Go        │ golang.org/x/net                    │ 0.0.0-20201021035429-f5854403a974 │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2022-0969        │ Go        │ golang.org/x/net                    │ 0.0.0-20201021035429-f5854403a974 │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2022-1144        │ Go        │ golang.org/x/net                    │ 0.0.0-20201021035429-f5854403a974 │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2022-0493        │ Go        │ golang.org/x/sys                    │ 0.0.0-20200930185726-fdedc70b468f │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2021-0113        │ Go        │ golang.org/x/text                   │ 0.3.3                             │ ../bnc-tendermint/go.mod │
│ https://osv.dev/vulnerability/GO-2022-1059        │ Go        │ golang.org/x/text                   │ 0.3.3                             │ ../bnc-tendermint/go.mod │
╰───────────────────────────────────────────────────┴───────────┴─────────────────────────────────────┴───────────────────────────────────┴──────────────────────────╯
```

### Example
The following commits are used to verify the compatibility：
original: https://github.com/j75689/bnc-tendermint/blob/cf1e338824f9409876b85527f38f1948effd505f/crypto/secp256k1/secp256k1_internal_test.go#L1019
new: https://github.com/j75689/bnc-tendermint/blob/34773fd8f16c06c6f4119b178bc5a8b23b2dc3f2/crypto/secp256k1/secp256k1_internal_test.go#L1019

### Changes

Notable changes:
* go.mod
* crypto/secp256k1

### CODE REVIEW

File: crypto/secp256k1/secp256k1.go [hunk 1]
```diff
@@ -37,25 +37,24 @@ func init() {
...
 
-// PrivKeySecp256k1 implements PrivKey.
-type PrivKeySecp256k1 [32]byte
+// PrivKey implements PrivKey.
+type PrivKeySecp256k1 []byte
 
 // Bytes marshalls the private key using amino encoding.
 func (privKey PrivKeySecp256k1) Bytes() []byte {
-	return cdc.MustMarshalBinaryBare(privKey)
+	return []byte(privKey)
```
Comment (guagualvcha) [inline crypto/secp256k1/secp256k1.go:47]: The return value is changed, so it is incompatible?

File: crypto/secp256k1/secp256k1.go [hunk 2]
```diff
@@ -37,25 +37,24 @@ func init() {
...
 }
 
-//-------------------------------------
-
 var _ crypto.PrivKey = PrivKeySecp256k1{}
 
-// PrivKeySecp256k1 implements PrivKey.
-type PrivKeySecp256k1 [32]byte
+// PrivKey implements PrivKey.
+type PrivKeySecp256k1 []byte
```
Comment (guagualvcha) [inline crypto/secp256k1/secp256k1.go:43]: what is the motivation to change from array to slice? Is it possible to have more than 32 or less than 32 bytes now?

File: crypto/secp256k1/secp256k1.go [hunk 3]
```diff
@@ -127,50 +135,58 @@ func GenPrivKeySecp256k1(secret []byte) PrivKeySecp256k1 {
...
-	bz, err := cdc.MarshalBinaryBare(pubKey)
-	if err != nil {
-		panic(err)
-	}
-	return bz
-}
-
-func (pubKey PubKeySecp256k1) String() string {
-	return fmt.Sprintf("PubKeySecp256k1{%X}", [PubKeySecp256k1Size]byte(pubKey))
+	return []byte(pubKey)
```
Comment (guagualvcha) [inline crypto/secp256k1/secp256k1.go:189]: Shall we keep the `String()` function?

Comment (guagualvcha) [inline crypto/secp256k1/secp256k1.go:189]: I think it is a incompatible change.

File: crypto/secp256k1/secp256k1.go [hunk 4]
```diff
@@ -179,3 +195,55 @@ func (pubKey PubKeySecp256k1) Equals(other crypto.PubKey) bool {
 	}
 	return false
 }
+
+// VerifyBytes verifies a signature of the form R || S.
+// It rejects signatures which are not in lower-S form.
+func (pubKey PubKeySecp256k1) VerifyBytes(msg []byte, sigStr []byte) bool {
```
Comment (guagualvcha) [inline crypto/secp256k1/secp256k1.go:223]: I think we should perform a test that the current version and previous is compatible, all the function and outputs should be compared.

**Review by guagualvcha — APPROVED** [review]

File: crypto/secp256k1/secp256k1_test.go [hunk 5]
```diff
@@ -94,12 +93,17 @@ func TestGenPrivKeySecp256k1(t *testing.T) {
...
+			"some long secret",
+			[]byte("We live in a society exquisitely dependent on science and technology, " +
+				"in which hardly anyone knows anything about science and technology."),
+		},
 		{"another seed used in cosmos tests #1", []byte{0}},
 		{"another seed used in cosmos tests #2", []byte("mySecret")},
 		{"another seed used in cosmos tests #3", []byte("")},
 	}
 	for _, tt := range tests {
+		tt := tt
```
Comment (HSG88) [inline crypto/secp256k1/secp256k1_test.go:106]: I believe this line isn't necessary, or if it is i'm wondering what is it for?

File: crypto/secp256k1/secp256k1.go [hunk 6]
```diff
@@ -4,25 +4,29 @@ import (
 	"bytes"
 	"crypto/sha256"
 	"crypto/subtle"
+	"errors"
```
Comment (brilliant-lx) [inline crypto/secp256k1/secp256k1.go:7]: `crypto/secp256k1/secp256k1.go` is used as `nocgo`, right?
In BSC side, `nocgo` is not used, BSC use  `secp256k1_nocgo.go`, which is removed in this PR, is it reasonable?

File: crypto/secp256k1/secp256k1_cgo.go [hunk 7]
```diff
@@ -1,23 +0,0 @@
-// +build libsecp256k1
```
Comment (brilliant-lx) [inline crypto/secp256k1/secp256k1_cgo.go:1]: crypto/secp256k1/secp256k1.go is used as nocgo, right?
In BSC side, nocgo is not used, BSC use secp256k1_nocgo.go, which is removed in this PR, is it reasonable?

**Pull request merged**